src/pages/[platform]/build-a-backend/add-aws-services/geo/amazon-location-sdk/index.mdx

```diff
@@ -0,0 +1,282 @@
+    @NonNull
+    @Override
+    public CoroutineContext getContext() {
+        return GlobalScope.INSTANCE.getCoroutineContext();
+    }
+
+    @Override
+    public void resumeWith(@NonNull Object resultOrException) {
+        Log.i("MyAmplifyApp", resultOrException.toString());
+    }
+});
+```
+
+</Block>
+<Block name="Kotlin">
+
+```kotlin
+import android.util.Log
+import aws.sdk.kotlin.services.location.LocationClient
+import aws.sdk.kotlin.services.location.model.ListMapsRequest
+import com.amplifyframework.core.Amplify
+```
+
+```kotlin
+// Obtain reference to the Amazon Location Service client
+val geoPlugin = Amplify.Geo.getPlugin("awsLocationGeoPlugin")
+val locationClient = geoPlugin.escapeHatch as LocationClient
+
+// Send a new request to the Location Maps endpoint directly using the client
+val request = ListMapsRequest {  }
+val response = locationClient.listMaps(request)
+Log.i("MyAmplifyApp", response.entries.toString())
+```
+
+</Block>
+</BlockSwitcher>
+
+### Documentation Resources
+
+* [How to manage Amazon Location Service resources through console](https://docs.aws.amazon.com/location/latest/developerguide/welcome.html)
+
+**Maps**
+* [Using Amazon Location Maps in your application](https://docs.aws.amazon.com/location/latest/developerguide/using-maps.html)
+* [Amazon Location Maps API Reference](https://docs.aws.amazon.com/location-maps/latest/APIReference/API_Operations.html)
+
+**Places**
+* [Searching place and geolocation data using Amazon Location](https://docs.aws.amazon.com/location/latest/developerguide/searching-for-places.html)
+* [Amazon Location Places API Reference](https://docs.aws.amazon.com/location-places/latest/APIReference/API_Operations.html)
+
+**Device Tracking**
+* [Managing your tracker resources](https://docs.aws.amazon.com/location/latest/developerguide/managing-trackers.html)
+* [Amazon Location Trackers API Reference](https://docs.aws.amazon.com/location-trackers/latest/APIReference/API_Operations.html)
+
+</InlineFilter>
+
+<InlineFilter filters={['swift']}>
+
+If you need functionality in the AWSLocation framework used by the Amplify Geo category that isn't available, we provide an escape hatch so you can reference it directly.
+
+<Callout warning>
+
+**Note:** While the Amplify Library for Swift is production ready, please note that the underlying AWS SDK for Swift is currently in Developer Preview, and is not yet intended for production workloads. [Here is additional reading material](https://github.com/awslabs/aws-sdk-swift/blob/main/Sources/Core/AWSSDKForSwift/Documentation.docc/stability.md) on the stability of the SDK
+
+</Callout>
+
+Note: If you provisioned your Geo resources via Amplify CLI, then the IAM policy will be specifically scoped to only allow actions required by the library. Please [adjust your authorization permissions](/[platform]/build-a-backend/more-features/geo/existing-resources/) accordingly for your escape hatch use-cases.
```

Comment:
this doesn't make sense anymore in Gen 2, right?

Reply:
thank you removed it